velox/exec/WindowPartition.cpp

```diff
@@ -270,8 +272,14 @@ void WindowPartition::updateKRangeFrameBounds(
     const vector_size_t* rawPeerBounds,
     vector_size_t* rawFrameBounds) const {
   column_index_t orderByColumn = sortKeyInfo_[0].first;
-  RowColumn frameRowColumn = data_->columnAt(frameColumn);
+  CompareFlags flags;
+  if constexpr (isAscending) {
+    flags.nullsFirst = sortKeyInfo_[0].second.isNullsFirst();
+  } else {
+    flags.nullsFirst = !sortKeyInfo_[0].second.isNullsFirst();
+  }
```

Comment:
Curious why we don't simply set `flags = {sortKeyInfo_[0].second.isNullsFirst(), isAscending, false}` but need this custom logic instead?

Reply:
@kagamiori : The actual data could be ascending or descending ordered (depending on ORDER BY). The range frame is trying to find a particular value in the ORDER BY column. But depending on preceding/following asc/desc we look for smallest value greater than or greatest value less than. The comparisons for this were simpler to write if the comparison test was with asc semantics only. So the comparison flags are always ascending, but the nulls flag needs to flip as well depending on asc/desc and how it compares with any value.

Reply:
I see. This makes sense to me. Thank you for explaining! Could we add a comment here with this explanation too?

Reply:
@kagamiori : As I played with the comparisons more, this code could be simplified keeping CompareFlags consistent with the ordering. I think the current code reads well with the current comments. But please let me know your suggestions.